### PR TITLE
Fix failing tests.

### PR DESCRIPTION
**Pull request by Youngestdev**
1. Updated `main.py`  to include a custom FastAPI class. In this class, I have added a `logger` attribute to fix the error from `mypy`.

2. Updated the `noxfile.py` to include the `--namespace-packages` tag to ensure `mypy` pursues the library path instead of a shallow search returning a module has no attribute error.
3. Fixed the wrong class import in the `service_result.py` file.
4. Created the `mypy.ini` configuration file to ignore missing imports.
5. Updated the `noxfile.py` file to clean the cache before running the `pre-commit test`. The cache is responsible for failed tests.
6. Added docstrings to `ModifiedFastAPI` class in `app/main.py`

### CODE REVIEW

**Comment by Youngestdev** [comment]
Once [this](https://github.com/geopython/pygeoapi/pull/894) is merged, all tests will pass :D.

**Comment by francbartoli** [comment]
Let's wait for the merge of https://github.com/geopython/pygeoapi/pull/894

**Comment by francbartoli** [comment]
Hi @Youngestdev, as a potential alternative can we pin click>8 on our `pyproject.toml` only? Or they are conflicting and so wouldn't work?

**Comment by Youngestdev** [comment]
It’s conflicting. We can instead pin to a fork/branch using the updates requirement. 

I can do that pending when the original issue is resolved.

**Comment by francbartoli** [comment]
ok, we can fork `pygeoapi` eventually and keep the change on our organization in a dedicated branch called `safety`.  Let's wait for the decision from the pygeoapi side

**Comment by Youngestdev** [comment]
Okay then. I’ll do that.

**Comment by francbartoli** [comment]
> Okay then. I’ll do that.

you don't have the right permissions to do that right now. I'll do the fork eventually and you will send the PR for the change so that we can then use the `safety` branch in this PR

File: app/main.py [hunk 1]
```diff
@@ -17,9 +20,16 @@
 from app.utils.request_exceptions import request_validation_exception_handler
 
 
-def create_app() -> FastAPI:
+class ModifiedFastAPI(FastAPI):
```
[the remark this answers is not in the document]
Comment (francbartoli) [reply app/main.py:22]: @Youngestdev can you send `ModifiedFastAPI` changed into `FastGeoAPI` in the meanwhile?

**Review by francbartoli — CHANGES_REQUESTED** [review]

**Comment by Youngestdev** [comment]
Sure. Give me a few minutes.

File: app/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Youngestdev) [reply app/main.py:22]: Sure.

[the remark this answers is not in the document]
Comment (Youngestdev) [reply app/main.py]: Done @francbartoli

**Comment by francbartoli** [comment]
some checks are now failing again @Youngestdev

**Comment by Youngestdev** [comment]
I'll take a look at them.

**Review by francbartoli — APPROVED** [review]

**Pull request merged**